src/vm/parser.rs

```diff
@@ -38,28 +44,30 @@ impl LexMatcher {
 
 fn get_value_or_err(input: &str, captures: Captures) -> Result<String> {
     let matched = captures.name("value").ok_or(
-        Error::ParseError("Failed to capture value from input".to_string()))?;
+        Error::new(ErrType::ParseError("Failed to capture value from input".to_string())))?;
     Ok(input[matched.start()..matched.end()].to_string())
 }
 
 pub fn lex(input: &str) -> Result<Vec<LexItem>> {
```

Comment:
This function is getting pretty big.  Maybe it can be split into a few smaller functions to make it easier to reason about?